src/app.ts

```diff
@@ -15,25 +17,30 @@ declare global {
 }
 
 interface SecretNotes {
-  notes: any[]
+  notes: Array<any>
 }
 
 const ceramicPromise = createCeramic()
 
 const authenticate = async (): Promise<string> => {
   const [ceramic, provider] = await Promise.all([ceramicPromise, getProvider()])
-  await ceramic.setDIDProvider(provider)
+  const did = new DID({
+    provider,
+    resolver: { ...KeyDidResolver.getResolver(), ...ThreeIdResolver.getResolver(ceramic) },
+  })
+  await did.authenticate()
+  window.did = did
+  ceramic.did = did
   const idx = createIDX(ceramic)
-  window.did = ceramic.did
   return idx.id
 }
 
 const ethAddressToDID = async (address: string): Promise<string> => {
   const caip10Doc = await window.ceramic?.createDocument('caip10-link', {
     metadata: {
       family: 'caip10-link',
-      controllers: [address.toLowerCase() + '@eip155:1']
-    }
+      controllers: [address.toLowerCase() + '@eip155:1'],
+    },
```

Comment:
Here the Caip10Link streamtype could be used.

Reply:
Is this needed for this PR please? The main goal was to use `@3id/connect` so I also had to updated the related dependencies but I didn't go through other parts of the logic.